### DISCUSSION TITLE
permanent storage

### DISCUSSION THREAD

**Discussion by octavei** [silius-rs/silius#170] [Q&A]
At present, our transaction pool uses dashmap as memory storage, but as a centralized bundler (currently), it actually needs to permanently store the transactions in the current transaction pool. It would be perfect if added to permanent storage.

**Comment by Vid201** [comment]
> At present, our transaction pool uses dashmap as memory storage, but as a centralized bundler (currently), it actually needs to permanently store the transactions in the current transaction pool. It would be perfect if added to permanent storage.

That's a good point. For the user operation pool, you can already use database implementation (https://github.com/Vid201/aa-bundler/blob/main/crates/uopool/src/database/mempool.rs) which stores the mempool on the disk (replace `MemoryMempool` here https://github.com/Vid201/aa-bundler/blob/main/crates/grpc/src/uopool.rs#L339). We don't have the same implementation for reputation part yet.

But it would make sense to switch default implementation to database instead of memory. Let me create issue for that.

**Reply by octavei** [reply]
Great job!
